service/cast.go

```diff
@@ -0,0 +1,49 @@
+package service
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Cast converts map[string]string to map[string]interface{} based on defined types in the service tasks map.
+func (s *Service) Cast(taskKey string, taskData map[string]string) (map[string]interface{}, error) {
+	task, ok := s.Tasks[taskKey]
+	if !ok {
+		return nil, fmt.Errorf("Task %q not found", taskKey)
+	}
+
+	m := make(map[string]interface{}, len(taskData))
+	for key, value := range taskData {
+		inputType, ok := task.Inputs[key]
+		if !ok {
+			return nil, fmt.Errorf("Task %q has no input named %q", taskKey, key)
```

Comment:
Please create a new type for this error in `service/error.go`

Reply:
done